### PR TITLE
Fixes #24974 support HardSwishInt8

### LINKED ISSUE #24974: TFLite importer: add support for "HardSwishInt8" activation layer
### Describe the feature and motivation

We usually use the "Hard-Swish" activation in our INT8-quantized models since is "friendly & cheap" to the inference backends used by multiple edge devices we are working on. Swish and other similar activations are already implemented for INT8-quantized models, so this addition should be "easy". TFLite importer already recognizes the "HARD_SWISH" layer "opcode" but doesn't implement it in INT8.

### Additional context

This feature addition is probably very straight-forward, where at least two changes should be performed:

1. In ["dnn/src/init.cpp"](https://github.com/opencv/opencv/blob/4.x/modules/dnn/src/init.cpp#L214), this line should be added:
`CV_DNN_REGISTER_LAYER_CLASS(HardSwishInt8,    ActivationLayerInt8);`

2. In https://github.com/opencv/opencv/blob/4.x/modules/dnn/src/tflite/tflite_importer.cpp#L940 , this line should be added (based on "Searching for MobileNetV3", 2019): 
`            else if (opcode == "HARD_SWISH")
                y = x * std::min(std::max(x + 3.f, 0.f), 6.f) * 0.166666667;`

I don't have an extensive context-awareness of the model construction mechanism in the DNN module so there might be something else that must be added/modified.

### PR DESCRIPTION
**Pull request by Dhanwanth1803**
As given very clearly in the issue #24974 I made the required 2 changes to implement HardSwish Layer in INT8. Requesting comments.
@dkurt 

resolves https://github.com/opencv/opencv/issues/24974

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [X] The PR is proposed to the proper branch
- [X] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by Dhanwanth1803** [comment]
@zihaomu Can you please check this pr and tell if any changes are required Sir.

**Comment by dkurt** [comment]
Thanks for the fix. Can you also add a line to OpenVINO backend at https://github.com/opencv/opencv/blob/76548e29bd62f299386cb919824b34ce4baf73cb/modules/dnn/src/int8layers/elementwise_layers.cpp#L249 ?

File: modules/dnn/src/tflite/tflite_importer.cpp [hunk 1]
```diff
@@ -939,6 +939,8 @@ void TFLiteImporter::parseActivation(const Operator& op, const std::string& opco
                 y = std::min(std::max(x, 0.f), 6.f);
             else if (opcode == "LOGISTIC")
                 y = 1.0f / (1.0f + std::exp(-x));
+            else if (opcode == "HARD_SWISH")
+                y = x * std::min(std::max(x + 3.f, 0.f), 6.f) * 0.166666667;
```
Comment (dkurt) [inline modules/dnn/src/tflite/tflite_importer.cpp:943]: Formula is different from https://github.com/opencv/opencv/blob/76548e29bd62f299386cb919824b34ce4baf73cb/modules/dnn/src/layers/elementwise_layers.cpp#L1900

**Comment by Dhanwanth1803** [comment]
> Thanks for the fix. Can you also add a line to OpenVINO backend at
> 
> https://github.com/opencv/opencv/blob/76548e29bd62f299386cb919824b34ce4baf73cb/modules/dnn/src/int8layers/elementwise_layers.cpp#L249
> 
> ?

Yes Sir. Will do.

**Comment by Dhanwanth1803** [comment]
@dkurt Sir, Can you look and tell me if more changes are required.

File: modules/dnn/src/int8layers/elementwise_layers.cpp [hunk 2]
```diff
@@ -267,6 +267,9 @@ class ActivationLayerInt8Impl CV_FINAL : public ActivationLayerInt8
             res = std::make_shared<ngraph::op::Elu>(input, 1.0f);
         } else if (type == "MishInt8") {
             res = std::make_shared<ngraph::op::v4::Mish>(input);
+        } else if (type == "HardSwishInt8") {
+            res = std::make_shared<ngraph::op::HardSwish>(input);
+        }
```
Comment (dkurt) [inline modules/dnn/src/int8layers/elementwise_layers.cpp:272]: incorrect syntax

Comment (Dhanwanth1803) [reply modules/dnn/src/int8layers/elementwise_layers.cpp]: My bad. Corrected sir.

**Review by dkurt — APPROVED** [review]
There is no test, but I think changes are simple.

**Comment by Dhanwanth1803** [comment]
> There is no test, but I think changes are simple.

So should I make many changes to the test file?

**Comment by Dhanwanth1803** [comment]
> There is no test, but I think changes are simple.

Sir, Can I know why the OpenVino test is failing in workflow runs?

**Comment by opencv-alalek** [comment]
> Sir, Can I know why the OpenVino test is failing in workflow runs?

There is build failure:

```
/home/openvino/opencv/modules/dnn/src/int8layers/elementwise_layers.cpp:271:48: error: 'HardSwish' is not a member of 'ngraph::op'
  271 |             res = std::make_shared<ngraph::op::HardSwish>(input);
      |                                                ^~~~~~~~~
```

Perhaps we need to guard this code by proper OpenVINO version check.

**Comment by dkurt** [comment]
Sorry, missed that. Should be `ngraph::op::v4::HSwish`

**Comment by Dhanwanth1803** [comment]
> Sorry, missed that. Should be `ngraph::op::v4::HSwish`

Will change it tonight Sir

**Comment by Dhanwanth1803** [comment]
@dkurt Sir, Am I missing something here?

**Pull request merged**